### ISSUE TITLE
why  PCS values are so small ?

### ISSUE THREAD

**Issue by donehii** [nicrie/xeofs#81]
I applied the EOF from xeofs module with S-mode, then I found that all the PCS are too small, between -1 and 1,  how can I get the PCS looks like a Normalized time series  as eofs module ?

**Comment by nicrie** [comment]
The PC scores $u$ that are returned by `xeofs` are **normalized** by default i.e. $|u|=1$.

What you're probably looking for are the **raw** or **unscaled** PC scores, which you obtain by projecting the data $X$ onto the normalized eigenvectors $V$. 

In xeofs you'd get them by multiplying your normalized scores with the square root of the eigenvalues $\lambda$ , so in practice:

```py
scores = model.scores()
expvar = model.explained_variance()
scores_scaled = scores * np.sqrt(expvar)
```
Let me know if that works for you

**Comment by tpdp** [comment]
Hi @nicrie , 

first of all, thank you very much for making such an awesome package available. I'm realling enjoying it.

I had the same issue here trying to reproduce a simple EOF analysis using Andrew Dawsons's eofs module, the one I had used in the past.

I tried to compute the un-scaled PC scores with the piece of code you provided and compare with the un-scaled option of eofs module, but results don't agree. 

When you say PC scores are normalized, you are saying that they are divided by their vector L2 norm? 

Indeed, when I compute un-scaled PCs using eofs module and divide by their norms (check code below), I get exactly the same PCs provided by default by xeofs.

Did I misunderstood something here? How do I get PCs un-scaled or scaled like the output options provided by eofs module [https://ajdawson.github.io/eofs/latest/api/eofs.standard.html](url)?

The code I'm using for these tests is below.

```python
from xeofs.models import EOF
from eofs.standard import Eof
import xarray as xr
import matplotlib.pyplot as plt
import numpy as np
from tempfile import gettempdir
TMP = gettempdir()

# -----------------------------------------------------------------------------
# Load example dataset
# -----------------------------------------------------------------------------
sst = xr.tutorial.open_dataset('ersstv5')['sst']

# -----------------------------------------------------------------------------
# 1) Comparing un-scaled PCs
# -----------------------------------------------------------------------------
# EOF using xeofs
# -----------------------------------------------------------------------------
model = EOF(n_modes=1, standardize=False)
model.fit(sst, dim='time')
scores = model.scores()
expvar = model.explained_variance()
scores_raw = scores * np.sqrt(expvar)

# -----------------------------------------------------------------------------
# EOF using Andrew Dawson's eofs module
# -----------------------------------------------------------------------------
model1 = Eof(sst.data)

# un-scaled PCs
scores0 = model1.pcs(npcs=1, pcscaling=0) 

# Plot
fig, ax = plt.subplots(figsize=(12, 4.8))
ax.plot(sst['time'], -scores0[:,0], label='eofs.standard')
ax.plot(sst['time'], scores_raw.sel(mode=1), ls='dashed', label='xeofs.EOF (raw)')
ax.legend()
plt.savefig(f'{TMP}/unscaled_pcs.jpeg')

# -----------------------------------------------------------------------------
# now, compare normalized PCs
# -----------------------------------------------------------------------------
# divide Andrew Dawsons's eofs PCs by thyer L2 norm
scores0_nrm = scores0 / np.linalg.norm(scores0, ord=2)

# Plot
fig, ax = plt.subplots(figsize=(12, 4.8))
ax.plot(sst['time'], -scores0_nrm[:,0], label='eofs.standard (norm)')
ax.plot(sst['time'], scores.sel(mode=1), ls='dashed', label='xeofs.EOF (norm)')
ax.legend()
plt.savefig(f'{TMP}/normalized_pcs.jpeg')
```

> The PC scores u that are returned by `xeofs` are **normalized** by default i.e. |u|=1.
> 
> What you're probably looking for are the **raw** or **unscaled** PC scores, which you obtain by projecting the data X onto the normalized eigenvectors V.
> 
> In xeofs you'd get them by multiplying your normalized scores with the square root of the eigenvalues λ , so in practice:
> 
> ```python
> scores = model.scores()
> expvar = model.explained_variance()
> scores_raw = scores * np.sqrt(expvar)
> ```
> 
> Let me know if that works for you

**Comment by nicrie** [comment]
Yes, you're absolutely right—it's indeed the L2 norm!

I apologize for the oversight. I failed to verify the default scaling of `eofs`. Contrary to what I mentioned earlier, the PCs in Dawson's package are scaled by the singular values, not the eigenvalues.

To elaborate a bit, there are three potential ways to scale the PCs. If you take your data matrix $X \in R^{n \times p}$, where you have $n$ samples and $p$ features, and you've derived $m$ PCs with $n$ time steps in a matrix $U \in R^{n \times m}$, then the three scaling methods would be:

1. $U$ such that $U^TU = \mathbb{1}$
2. $U\Lambda$, with $\Lambda$ as a diagonal matrix containing the eigenvalues of the covariance matrix of $X$
3. $US$, where $S$ is a diagonal matrix holding the singular values of $X$

Notably, while `eofs` employs the third scaling, `xeofs` adopts the first.

---

### In a Nutshell

```py
scores = model.scores()
svals = model.singular_values()
scores_raw = scores * svals
```
This will yield the scaling consistent with `eofs`.

And a side note: I appreciate you flagging this—it's a reminder that I should clarify this aspect in the documentation.

**Comment by tpdp** [comment]
Thank you for the blazingly fast, clear and concise response @nicrie. Everything makes sense now and I can get scalings consistent accross both packages.

**Comment by nicrie** [comment]
Fantastic! I leave this open as a reminder for me to improve the documentation

**Comment by donehii** [comment]
> The PC scores u that are returned by `xeofs` are **normalized** by default i.e. |u|=1.
> 
> What you're probably looking for are the **raw** or **unscaled** PC scores, which you obtain by projecting the data X onto the normalized eigenvectors V.
> 
> In xeofs you'd get them by multiplying your normalized scores with the square root of the eigenvalues λ , so in practice:
> 
> ```python
> scores = model.scores()
> expvar = model.explained_variance()
> scores_raw = scores * np.sqrt(expvar)
> ```
> 
> Let me know if that works for you

Thanks for your responses! After reviewing your communication with tpdq, I think I have found my answer, then I'll try  again about this issue. I really appreciate all of you gays. Then, there is another question, would mind to tell me  what 's the meaning of  L2 norm, or how can I find the introuduction about it ?

**Comment by nicrie** [comment]
This is now explained in the [Quickstart](https://xeofs.readthedocs.io/en/latest/quickstart.html) section.
